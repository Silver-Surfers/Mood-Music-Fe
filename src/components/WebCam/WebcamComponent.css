:root {
  --red: #e63946ff;
  --violet: #6f58c9ff;
  --purple: #3c153bff;
  --orange: #ff6700ff;
  --green: #1DB954;
  --yellow: #e5f77dff;
  --blue:#457b9dff;
  --neutral: #f1faeeff;
}

.bigBox {
  display: flex;
  flex-direction: column;
  align-items: center;
  background: linear-gradient(45deg, var(--neutral), var(--green), var(--neutral), var(--yellow));
  min-height: 100vh;
}

.webcam {
    margin-top: 40px;
    margin-bottom: 20px;
    width: 80vw;
    border: solid black 5px;
    border-radius: 100px;
    box-shadow: gray 5px 5px 5px;
}
.button {
  display: flex;
  align-items: center;
  justify-content: center;

  min-width: 100px;
  height: 40px;
  text-transform: uppercase;
  margin-top: 10px;
  letter-spacing: 1.1px;
  font-weight: 600;
  background: linear-gradient(90deg, rgba(101,211,110,1) 0%, rgb(10, 110, 55) 100%);
  border: solid black 3px;
  border-radius: 100px;
  padding: 5px;
  box-shadow: gray 3px 3px 5px;
}
