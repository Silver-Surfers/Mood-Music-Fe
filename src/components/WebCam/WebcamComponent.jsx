import React, { useEffect, useState, useRef, useCallback } from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import Webcam from 'react-webcam';
import { fetchImage, setImage } from '../../actions/azureActions';
import { setToken } from '../../actions/spotifyActions';
import PropTypes from 'prop-types';
import styles from './WebcamComponent.css';


export const WebcamCapture = ({ location }) => {
  const webcamRef = useRef(null);
  const [imgSrc, setImgSrc] = useState(null);
  const history = useHistory();
  const dispatch = useDispatch();

  const { search } = location;
  const params = new URLSearchParams(search);
  const token = params.get('token');
  console.log(token);

  const capture = useCallback(async() => {
    const imageSrc = webcamRef.current.getScreenshot();
    setImgSrc(imageSrc);
    await dispatch(setImage(imageSrc));
    history.push('/image');
  }, [webcamRef, setImage]);

  useEffect(() => {
    if(token) dispatch(setToken(token));

  }, []);

  useEffect(async() => {
   
    if(!imgSrc) return;
    const res = await fetch(imgSrc);
    const blob = await res.blob();
    // const arr = await blob.arrayBuffer();

    dispatch(fetchImage(blob));
  
  }, [imgSrc]);


  
  return (
    
    <div className={styles.bigBox}>
      <Webcam

        className={styles.webcam}
        audio={false}
        ref={webcamRef}
        screenshotFormat="image/jpeg"
      />
      <button onClick={capture}>Capture photo</button>
    
    </div>
  );
};

WebcamCapture.propTypes = {
  location: PropTypes.object.isRequired
};
