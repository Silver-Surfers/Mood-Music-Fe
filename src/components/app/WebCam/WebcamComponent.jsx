import React, { useEffect, useState, useRef, useCallback } from 'react';
import { useDispatch } from 'react-redux';
import Webcam from 'react-webcam';
import { fetchImage } from '../../../actions/azureActions';

export const WebcamCapture = () => {
  const webcamRef = useRef(null);
  const [imgSrc, setImgSrc] = useState(null);
  const dispatch = useDispatch();

  const capture = useCallback(() => {
    const imageSrc = webcamRef.current.getScreenshot();
    setImgSrc(imageSrc);
  }, [webcamRef, setImgSrc]);

  useEffect(async() => {
    if(!imgSrc) return;
    const res = await fetch(imgSrc);
    const blob = await res.blob();
    const arr = await blob.arrayBuffer();
    console.log(blob);
    console.log(arr);
    dispatch(fetchImage(blob));
  }, [imgSrc]);
  
  return (
    <>
      <Webcam
        audio={false}
        ref={webcamRef}
        screenshotFormat="image/jpeg"
      />
      <button onClick={capture}>Capture photo</button>
      {imgSrc && (
        <img
          src={imgSrc}
        />
      )}
    </>
  );
};
