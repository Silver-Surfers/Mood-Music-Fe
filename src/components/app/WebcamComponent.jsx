import React, { useEffect, useState, useRef, useCallback } from 'react';
import Webcam from 'react-webcam';
import ReactDOM from 'react-dom';
import { makeRequest } from '../../services/mood-music-api';

export const WebcamCapture = () => {
  const webcamRef = useRef(null);
  const [imgSrc, setImgSrc] = useState(null);
  
  const capture = useCallback(() => {
    const imageSrc = webcamRef.current.getScreenshot();
    setImgSrc(imageSrc);
  }, [webcamRef, setImgSrc]);

  useEffect(async() => {
    if(!imgSrc) return;
    const res = await fetch(imgSrc);
    const blob = await res.blob();
    const arr = await blob.arrayBuffer();
    console.log(blob);
    console.log(arr);
    await makeRequest(arr);
  }, [imgSrc]);
  
  return (
    <>
      <Webcam
        audio={false}
        ref={webcamRef}
        screenshotFormat="image/jpeg"
      />
      <button onClick={capture}>Capture photo</button>
      {imgSrc && (
        <img
          src={imgSrc}
        />
      )}
    </>
  );
};
  
ReactDOM.render(<WebcamCapture />, document.getElementById('root'));
