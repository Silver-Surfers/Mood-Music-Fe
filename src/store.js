import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';


import reducer from './reducers/index';


 284730e237e0166107f72624f51fa5df8a4e0523

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
export default createStore(
  reducer,
  composeEnhancers(applyMiddleware(thunk))
);
