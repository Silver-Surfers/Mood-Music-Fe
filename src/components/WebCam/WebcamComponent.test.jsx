import React from 'react';
import { 
  MemoryRouter as Router,
  Route, 
  Switch
} from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import { WebcamCapture } from './WebcamComponent.jsx';
import { Provider } from 'react-redux';
import store from '../../store';

describe('Webcam component', () => {
  afterEach(() => cleanup());
  it('renders takes a snapshot photo through the user/s webcam', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router 
          initialEntries={['/webcam']}>
          <Switch>
            <Route
              exact path="/webcam"
              component={WebcamCapture}
            />
          </Switch>
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
