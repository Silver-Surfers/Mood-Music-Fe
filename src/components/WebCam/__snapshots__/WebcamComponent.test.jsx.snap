// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webcam component renders takes a snapshot photo through the user/s webcam 1`] = `
<DocumentFragment>
  <div
    class="bigBox"
  >
    <video
      autoplay=""
      class="webcam"
      playsinline=""
    />
    <button>
      Capture photo
    </button>
  </div>
</DocumentFragment>
`;
